service/api_logic/streams_logic.py

```diff
@@ -1,126 +1,48 @@
             Stream.stream_id,
-            Stream.stream_url,
+            Stream.title,
             Stream.start_time,
             Stream.sport_id,
-            StreamsStatus.status_id
-        )
-        .join(StreamsStatus, Stream.stream_id == StreamsStatus.stream_id)
-    )
-
-    today_start = datetime.combine(datetime.date.today(), datetime.min.time())
-    today_end = datetime.combine(datetime.date.today(), datetime.max.time())
-    query = query.filter(
-        Stream.start_time.between(today_start.timestamp(), today_end.timestamp())
-    )
-
-
-    if pagination:
-        query = query.offset(pagination.offset).limit(pagination.limit)
-
-    return query.all()
-
-
-@handle_exceptions
-@logger.log_function_call()
-def save_json_stream_to_streams_table(session, streams_data):
-    streams_list = json.loads(streams_data) if isinstance(streams_data, str) else streams_data
-
-    for stream_data in streams_list:
-        stream = Stream(
-                stream_url=stream_data["stream_url"],
-                sport_id=stream_data["sport_id"],
+            func.array_agg(StreamUrl.stream_url).label('stream_url')
         )
-
-        session.merge(stream)  
-    session.commit()
-
-@handle_exceptions
-@logger.log_function_call()
-def save_json_stream_to_status_table(session,streams_data):
-    streams_list = json.loads(streams_data) if isinstance(streams_data, str) else streams_data
-
-    for stream_data in streams_list:
-        stream = Streams_Status(
-            start_time=stream_data["time"],
-            status=stream_data["status"]
-
+         .join(StreamUrl, Stream.stream_id == StreamUrl.stream_id)
+         .group_by(Stream.stream_id)
         )
 
-        session.merge(stream)
-    
-    session.commit()
-
-
-
-
-@handle_exceptions
-@logger.log_function_call()
-def get_streams_by_count(session, count:int):
-    streams = fetch_streams(session, order_by=Stream.start_time.desc(), limit=count)
-    return json_streams(streams)
-
-@handle_exceptions
-@logger.log_function_call()
-def get_latest_sport_streams(session, count:int, sport_name:str):
-    sport = get_sport_index_by_name(session, sport_name)
-    filters = [Stream.sport_id == sport.sport_id]
-    streams = fetch_streams(session, order_by=Stream.start_time.desc(), limit=count, filters=filters)
-    return json_streams(streams)
+        filtered_query, count = FilterManagerStrategy.apply_filters(Stream, query, filters_dto)
 
-@handle_exceptions
-@logger.log_function_call()
-def get_streams_of_sport(session, sport_name:str):
-    sport = get_sport_index_by_name(session, sport_name)
-    filters = [Stream.sport_id == sport.sport_id]
-    streams = fetch_streams(session, order_by=Stream.start_time.desc(), filters=filters)
-    return json_streams(streams)
+        streams = self._stream_dal.query_output(filtered_query)
+        stream_output = StreamsOutput(many=True)
+        streams = stream_output.dump(streams)
```

Comment:
We can just do that in one line:

```suggestion
        stream_output = StreamsOutput(many=True).dump(streams)
```